core/modules/file/file.module

```diff
@@ -1621,6 +1645,110 @@ function file_managed_file_pre_render($element) {
   return $element;
 }
 
+/**
+ * AJAX callback for managed_file elements to open a browser modal dialog.
+ *
+ * @see file_managed_file_process()
+ */
+function file_managed_file_browser_open($form, $form_state) {
+  // Get the parent "managed_file" element.
+  $parents = $form_state['triggering_element']['#array_parents'];
+  array_pop($parents);
+  $element = backdrop_array_get_nested_value($form, $parents);
+
+  // Pull the view to open off of the managed_file #browser_view property.
+  $view_name = $element['#browser_view'];
+  $view = views_embed_view($view_name);
+
+  $submit_form = backdrop_get_form('file_managed_file_browser_form');
+
+  $title = t('Select from Image Library');
+  $options = array(
+    'dialogClass' => 'image-library-container',
+    'width' => '90%',
+  );
+  $html = '';
+  $html .= theme('status_messages');
+  $html .= '<div class="image-library">';
+  $html .= '<div class="image-library-view">';
+  $html .= $view;
+  $html .= backdrop_render($submit_form);
+  $html .= '</div><div class="image-detail"><h3>Image details</h3>';
+  // Make placeholders for properties to be added by file.js
+  $html .= '<p>ID: <span class="image-fid"> ......</span></p>';
+  $html .= '<p>Name: <span class="image-name"> ......</span></p>';
+  $html .= '<p>Src: <span class="image-uri"> ......</span></p>';
+  $html .= '<p>Size: <span class="image-size"> ......</span></p>';
+  // Provide name of image field to be used by file.js.
+  $html .= '<span>Field name: <span class="field-name" .....</span></p>';
+  $html .= '<p><i>Point at an image to reveal its properties.</i></p>';
+  $html .= '<p><i>Click an image to select it.</i></p>';
+  $html .= '</div>'; // end of image-detail.
+  $html .= '</div>';
+
+  $settings = array(
+    'file' => array(
+      'browser' => array(
+        'currentFidElement' => '[name="' . $element['fid']['#name'] . '"]',
+      ),
+    ),
+  );
+
+  $commands = array();
+  $commands[] = ajax_command_settings($settings, TRUE);
+  $commands[] = ajax_command_open_modal_dialog($title, $html, $options);
+  return array('#type' => 'ajax', '#commands' => $commands);
+}
+
+/**
+ * Form to submit a file selected in the file browser.
+ */
+function file_managed_file_browser_form($form, &$form_state) {
+  $form['fid'] = array(
+    '#type' => 'hidden',
```

Comment:
Perhaps add a comment here as to why we are using `'#type' => 'hidden'` instead of `'#type' => 'value'`?

Reply:
Comment added.